daffodil-core/src/main/scala/org/apache/daffodil/dsom/ElementBase.scala

```diff
@@ -653,29 +653,29 @@ trait ElementBase
     }
   }
 
+  private lazy val optionBinaryNubmerRep = findPropertyOption("binaryNumberRep")
+  /* All binary number representations other than "binary" are of some packed format */
+  private lazy val isPackedBinaryNumber: Boolean = optionBinaryNumberRep.isDefined && (binaryNumberRep != BinaryNumberRep.Binary)
```

Comment:
I have updated the code to not rely on whether or not binaryNumberRep is defined.
I have also tried to tidy up the areas with all the pipes, but I still prefer the use of pipes instead of having several more case statements that evaluate to the same thing.